sentry_streams/src/filter_step.rs

```diff
@@ -57,24 +58,30 @@ impl ProcessingStrategy<RoutedValue> for Filter {
                 match py_res {
                     Ok(boolean) => {
                         let filtered = boolean.is_truthy(py).unwrap();
-                        if filtered {
-                            let _ = self.next_step.submit(message);
-                        }
-                        Ok(())
+                        Ok(filtered)
                     }
-                    Err(_) => match message.inner_message {
-                        InnerMessage::BrokerMessage(inner) => {
-                            Err(SubmitError::<RoutedValue>::InvalidMessage(InvalidMessage {
-                                partition: inner.partition,
-                                offset: inner.offset,
-                            }))
-                        }
-                        InnerMessage::AnyMessage(inner) => {
-                            panic!("Unexpected message type: {:?}", inner)
-                        }
-                    },
+                    Err(e) => return Err(e),
                 }
-            })
+            });
+            match res {
+                Ok(bool) => {
+                    if bool {
+                        let _ = self.next_step.submit(message);
```

Comment:
totally unrelated but it seems to me that backpressure is not handled correctly here
```suggestion
                        self.next_step.submit(message)?;
```

Reply:
fixed, thanks